docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc

```diff
@@ -1,12 +1,105 @@
+the specified interval includes the change to or from daylight savings time, the
+interval will end an hour sooner or later than you expect.
+
+There are similar differences to consider when you specify single versus multiple minutes or hours. Multiple time periods longer than a day are not supported.
+
+Here are the valid time specifications and their meanings:
+
+*milliseconds (ms)* +
+Fixed length interval; supports multiples.
+
+*seconds (s)* +
+1000 milliseconds; fixed length interval (except for the last second of a
+minute that contains a leap-second, which is 2000ms long); supports multiples.
+
+*minutes (m)* +
+All minutes begin at 00 seconds. +
+* One minute (1m) is the interval between 00 seconds of the first minute and 00
+seconds of the following minute in the specified timezone, compensating for any
+intervening leap seconds, so that the number of minutes and seconds past the 
+hour is the same at the start and end.
+* Multiple minutes (_n_m) are intervals of exactly 60x1000=60,000 milliseconds
+each.
+
+*hours (h)* +
+All hours begin at 00 minutes and 00 seconds. +
+* One hour (1h) is the interval between 00:00 minutes of the first hour and 00:00
+minutes of the following hour in the specified timezone, compensating for any
+intervening leap seconds, so that the number of minutes and seconds past the hour is the same at the start and end.
+* Multiple hours (_n_h) are intervals of exactly 60x60x1000=3,600,000 milliseconds each.
+
+*days (d)* +
+All days begin at the earliest possible time, which is usually 00:00:00
+(midnight).
+* One day (1d) is the interval between 00:00:00 of the first day and 00:00:00
+of the following day in the specified timezone, compensating for any intervening
+time changes, so that the number of hours, minutes, and seconds of the day is
+the same at the start and end.
+* Multiple days (_n_d) are intervals of exactly 24x60x60x1000=86,400,000 milliseconds each.
+
+*weeks (w)* +
+* One week (1w) is the interval between the start day:hour:minute:second and
+the same day and time of the following week in the specified timezone, so that
+the day and time are the same at the start and end.
+* Multiple weeks (_n_w) are not supported.
+
+*months (m)* +
+* One month (1m) is the interval between the start date and time and the same date and time of the following month in the specified timezone, so that the date and time are the same at the start and end.
+* Multiple months (_n_m) are not supported.
+
+*years (y)* +
+* One year (1y) is the interval between the start month, date, and time and the same
+month, date, and time the following year in the specified timezone, so that the date and time are the same at the start and end.
+* Multiple years (_n_y) are not supported.
+
+*quarters (q)* +
+* One quarter (1q) is the interval between the start month, date, and time and the same date and time three months later, so that the date and time are the
```

Comment:
Again I think we need to use "day of month" here to be unambiguous.

We should also correct the wrapping here.

Also can we move this above the years section? its a small thing but I like it ordered by increasing intervals 😄

Reply:
done